lib/index.ts

```diff
@@ -183,13 +204,20 @@ function collectDiagnostics(program: ts.Program) {
   });
 }
 
+interface ICompilerOptions {
```

Comment:
I suggest moving this to the top (since it's used earlier in the file), and also perhaps making these options generic? That would allow using `{}` as the default options, and only specifying one of them when one is to be overridden.

Reply:
What you mean we generic? Using {} looks more like a JavaScript solution not Typescript.

Reply:
Sorry, typo -- I meant "optional" :)

Reply:
The options are optional, there is a default value.